### DISCUSSION TITLE
Fluidsynth appears to not receive midi from an external controller

### DISCUSSION THREAD

**Discussion by dchp175** [FluidSynth/fluidsynth#1207] [Q&A]
Hello! First time posting here, and I'd like to ask for help with a somewhat specific situation:

I am currently trying to use a custom soundfont of mine (version 2.01, made in Polyphone 2.2.0) within Fluidsynth 1.0.3 running on Windows 2000 (32-bit). The issue is that while it seems to work perfectly fine with noteon/noteoff commands and acceptably well when playing midi files, I cannot get it to play anything with an external keyboard. Other programs do receive midi data from it and play normally.
A bit of troubleshooting (mostly with synth.verbose) led me to believe that, somehow, fluidsynth is not "connected" to the MIDI input port; with other programs, there usually is some sort of setting to choose the input and output, but fluidsynth doesn't give me any options like that and its official documentation doesn't seem to mention such a thing either.
The issue stays regardless of the loaded soundfont(s) or if the keyboard is connected with a MIDI or USB cable. I also started experiencing it on my main computer (fluid 2.1.6, win10 64b) with just about the same symptoms (I should also mention that I tried reinstalling both instances and nothing changed). I feel like the root of the problem is something very basic about how Windows handles MIDI, but neither the online documentation nor my "Windows common sense" have been of any help so far.

**Comment by dchp175** [comment]
Update: It turned out that the problem was caused by the game port adapter I was using to connect my MIDI keyboard. Fluidsynth does not receive any input from that, but works just fine with my EES PC-MIDI 2/4 parallel port adapter. I would nevertheless be grateful for any suggestions on how to fix the game port.
The actual sounds produced are still very much uneven in terms of volume, but I expect to be able to fix that after taking a proper look at it in a soundfont editor (it seems like there are compatibility issues between the new soundfont and old software).

**Comment by dflat2** [comment]
I had the exact same issue. FluidSynth was the only program unable to receive any MIDI input from my controller. I have no idea how it works in details, but the `-j` option (`--connect-jack-outputs`) fixed the problem.
